nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,483 @@
+"""
+Better brain parcellations for Region of Interest analysis
+"""
+
+import nibabel
+import numpy as np
+
+from scipy.ndimage import label
+from scipy.stats import scoreatpercentile
+
+from sklearn.base import clone
+from sklearn.externals.joblib import Memory
+
+from .. import masking
+from ..input_data import NiftiMasker, NiftiMapsMasker
+from ..input_data.nifti_masker import filter_and_mask
+from .._utils import check_niimg, check_niimg_3d, check_niimg_4d
+from .._utils.extmath import fast_abs_percentile
+from .._utils.class_inspect import get_params
+from ..image import iter_img, new_img_like
+from ..image.image import _smooth_array
+from .._utils.niimg_conversions import concat_niimgs
+from .._utils.compat import _basestring
+from ..externals.skimage import peak_local_max, random_walker
+
+
+def estimate_apply_threshold_to_maps(maps_img, mask_img=None, parameters=None,
+                                     threshold='auto',
+                                     estimate_threshold_value='percentile'):
```

Comment:
Before it was ```threshold_strategy```. I changed this name to be more informative since we estimate certain threshold value from the voxels or data. May be I stick to old one ```thresholding_strategy```.